packages/dataviews/src/field-types/email.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * WordPress dependencies
+ */
+import { isEmail } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import type {
+	DataViewRenderFieldProps,
+	SortDirection,
+	ValidationContext,
+	Operator,
+} from '../types';
+import { renderFromElements } from '../utils';
+import { OPERATOR_IS_ANY, OPERATOR_IS_NONE } from '../constants';
+
+function sort( valueA: any, valueB: any, direction: SortDirection ) {
+	return direction === 'asc'
+		? valueA.localeCompare( valueB )
+		: valueB.localeCompare( valueA );
+}
+
+function isValid( value: any, context?: ValidationContext ) {
+	// TODO: this implicitly means the value is required.
+	if ( value === '' ) {
+		return false;
+	}
+
+	if ( ! isEmail( value ) ) {
+		return false;
+	}
+
+	if ( context?.elements ) {
```

Comment:
Not specifically for this PR but this check seems like something common to all field types, I wonder if we should move it elsewhere.

Reply:
Try to remove this by https://github.com/Automattic/wp-calypso/pull/103579/commits/a9417aa6f1a1f409b97fea4a10b6b909874e5c2a. What do you think?

Reply:
Not sure how I feel about that. Let's keep that for its own PR. My initial thinking is that I imagined this to be part of the "validation framework" rather than having a "base" field type.